src/new/wallet.js

```diff
@@ -577,18 +531,17 @@ class HathorWallet extends EventEmitter {
         // So we will enqueue this message to be processed later
         this.wsTxQueue.enqueue(wsData);
       } else {
-        this.onNewTx(wsData);
+        this.newTxPromise = this.newTxPromise.then(() => {
```

Comment:
Whenever a new tx arrives on the websocket we need to process the storage metadata (similar to the old `preProcessedData`), so if a new transaction arrives during this process it may be possible that its promise finishes before the first one and when they both finish the metadata will be wrong.

To mitigate this, we wait for any call to `onNewTx` process to end before calling again with a promise chain.